File: test/merkletreejs/merkle_root/merkle_root_test.py
from merkly.mtree import MerkleTree
import hashlib
import json


def sha256(x, y):
    data = x + y
    return hashlib.sha256(data).digest()


leaves = ["a", "b", "c", "d"]
leaves = list(map(lambda x: x.encode(),leaves))
tree = MerkleTree(leaves, sha256)
root = tree.root.hex()

print(json.dumps({"root": root}))

File: test/merkletreejs/merkle_root/merkle_root_test.js
const { MerkleTree } = require('merkletreejs');
const SHA256 = require('crypto-js/sha256');

const leaves = ['a', 'b', 'c', 'd'].map(SHA256);
const tree = new MerkleTree(leaves, SHA256, {});
const root = tree.getRoot().toString('hex');

console.log(JSON.stringify({ root }));
